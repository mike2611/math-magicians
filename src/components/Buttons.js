import React,{ Component } from 'react';

export default class Buttons extends Component {
  constructor() {
    super();
    this.calculateHandler = this.calculateHandler.bind(this);
  }

  calculateHandler = (e) => {
    this.props.calculateHandler(e.target.innerText);
  }

  render() {
    return (
      <div className="calculator-btns">
        <button type="button" className="btn" onClick={this.calculateHandler}>AC</button>
        <button type="button" className="btn" onClick={this.calculateHandler}>+/-</button>
        <button type="button" className="btn" onClick={this.calculateHandler}>%</button>
        <button type="button" className="btn btn-orange" onClick={this.calculateHandler}>รท</button>
        <button type="button" className="btn" onClick={this.calculateHandler}>7</button>
        <button type="button" className="btn" onClick={this.calculateHandler}>8</button>
        <button type="button" className="btn" onClick={this.calculateHandler}>9</button>
        <button type="button" className="btn btn-orange" onClick={this.calculateHandler}>x</button>
        <button type="button" className="btn" onClick={this.calculateHandler}>4</button>
        <button type="button" className="btn" onClick={this.calculateHandler}>5</button>
        <button type="button" className="btn" onClick={this.calculateHandler}>6</button>
        <button type="button" className="btn btn-orange" onClick={this.calculateHandler}>-</button>
        <button type="button" className="btn" onClick={this.calculateHandler}>1</button>
        <button type="button" className="btn" onClick={this.calculateHandler}>2</button>
        <button type="button" className="btn" onClick={this.calculateHandler}>3</button>
        <button type="button" className="btn btn-orange" onClick={this.calculateHandler}>+</button>
        <button type="button" className="btn zero" onClick={this.calculateHandler}>0</button>
        <button type="button" className="btn" onClick={this.calculateHandler}>.</button>
        <button type="button" className="btn btn-orange" onClick={this.calculateHandler}>=</button>
      </div>
    );
  }
}
