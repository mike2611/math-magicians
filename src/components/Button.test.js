import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import Buttons from './Buttons';

describe('Buttons component', () => {
  test('should render Buttons component correctly', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<Buttons />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  test('Clicking a buttons and testing the inner HTML', () => {
    render(<Buttons />);

    const buttonOne = screen.queryByText(/AC/i);

    expect(buttonOne).toBeInTheDocument();
  });
});
