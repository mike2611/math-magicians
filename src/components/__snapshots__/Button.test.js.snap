// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons component should render Buttons component correctly 1`] = `
<div
  className="calculator-btns"
>
  <button
    className="btn"
    type="button"
  >
    AC
  </button>
  <button
    className="btn"
    type="button"
  >
    +/-
  </button>
  <button
    className="btn"
    type="button"
  >
    %
  </button>
  <button
    className="btn btn-orange"
    type="button"
  >
    รท
  </button>
  <button
    className="btn"
    type="button"
  >
    7
  </button>
  <button
    className="btn"
    type="button"
  >
    8
  </button>
  <button
    className="btn"
    type="button"
  >
    9
  </button>
  <button
    className="btn btn-orange"
    type="button"
  >
    x
  </button>
  <button
    className="btn"
    type="button"
  >
    4
  </button>
  <button
    className="btn"
    type="button"
  >
    5
  </button>
  <button
    className="btn"
    type="button"
  >
    6
  </button>
  <button
    className="btn btn-orange"
    type="button"
  >
    -
  </button>
  <button
    className="btn"
    type="button"
  >
    1
  </button>
  <button
    className="btn"
    type="button"
  >
    2
  </button>
  <button
    className="btn"
    type="button"
  >
    3
  </button>
  <button
    className="btn btn-orange"
    type="button"
  >
    +
  </button>
  <button
    className="btn zero"
    type="button"
  >
    0
  </button>
  <button
    className="btn"
    type="button"
  >
    .
  </button>
  <button
    className="btn btn-orange"
    type="button"
  >
    =
  </button>
</div>
`;
